src/storage-client/src/statistics.rs

```diff
@@ -464,6 +464,26 @@ pub struct SourceStatisticsUpdate {
     pub offset_committed: Gauge<Total>,
 }
 
+impl From<GlobalId> for SourceStatisticsUpdate {
+    fn from(id: GlobalId) -> SourceStatisticsUpdate {
```

Comment:
Using `From` for this feels a bit weird. It's not really that `GlobalId` is converted into a `SourceStatisticsUpdate` in the same way that a `usize` is not converted into a `Vec` when it goes through `with_capacity`. I suggest making these `new` methods